vcf-dumper/vcf-dumper-ws/src/main/java/uk/ac/ebi/eva/vcfdump/server/HtsgetVcfController.java

```diff
@@ -0,0 +1,235 @@
+        }
+
+        String dbName = "eva_" + species;
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<>();
+        queryParameters.put(VariantDBAdaptor.REGION, Collections
+                .singletonList(referenceName));
+
+        int blockSize = Integer.parseInt(evaProperties.getProperty("eva.htsget.blocksize"));
+
+        VariantExporterController controller = new VariantExporterController(dbName, Arrays.asList(id.split(",")),
+                                                                             evaProperties,
+                                                                             queryParameters, blockSize);
+        if (!controller.validateSpecies()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidInput", "The requested species is not available"));
+        }
+        if (!controller.validateStudies()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidInput", "The requested study(ies) is not available"));
+        }
+
+        if (start == null) {
+            start = controller.getCoordinateOfFirstVariant(referenceName);
+        }
+        if (end == null) {
+            end = controller.getCoordinateOfLastVariant(referenceName);
+        }
+
+        if (end <= start) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(new HtsGetError("InvalidRange", "The requested range cannot be satisfied"));
+        }
+
+        List<Region> regionList = controller.divideChromosomeInChunks(referenceName, start, end);
+
+        HtsGetResponse htsGetResponse = buildHtsGetResponse(id, referenceName, species, request, regionList);
+        return ResponseEntity.status(HttpStatus.OK).body(htsGetResponse);
+
+    }
+
+    private HtsGetResponse buildHtsGetResponse(String id, String referenceName, String species,
+                                               HttpServletRequest request, List<Region> regionList) {
+        HtsGetResponse htsGetResponse = new HtsGetResponse();
+        htsGetResponse.setFormat(VCF);
+        htsGetResponse.constructUrls(request.getLocalName() + ":" + request.getLocalPort(), id, referenceName, species,
+                regionList);
+        return htsGetResponse;
+    }
+
+
+    @RequestMapping(value = "/headers", method = RequestMethod.GET, produces = "application/octet-stream")
+    public StreamingResponseBody getHtsgetHeaders(
+            @RequestParam(name = "species") String species,
+            @RequestParam(name = "studies") List<String> studies,
+            HttpServletResponse response)
+            throws IllegalAccessException, IllegalOpenCGACredentialsException, InstantiationException, IOException,
+            StorageManagerException, URISyntaxException, ClassNotFoundException {
+
```

Comment:
Remove blank lines in this method, it is short enough to be understandable without spacing.